egs/wsj/s5/steps/nnet3/ctc/train_lstm.sh

```diff
@@ -238,12 +242,38 @@ if  [ $stage -le -6 ]; then
   echo "$0: creating CTC transition model"
 
   num_phones=$(cat $lang/phones.txt | grep -v '^#' | tail -n +2 | wc -l) || exit 1;
-  # important not to mak
-  $cmd $dir/log/init_trans_model.log \
-    ctc-init-transition-model --num-phones=$num_phones \
-       $alidir/tree \
+  num_cutoffs=$(echo $cutoffs|wc -w)
+  rm -f $dir/0.*.ctc_trans_mdl
+  $cmd JOB=1:$num_cutoffs $dir/log/init_trans_model.JOB.log \
+    cutoff_array=\(0 $cutoffs\) '&&' \
+    ctc-init-transition-model  --state-count-cutoff2plus=\${cutoff_array[JOB]} \
+       --num-phones=$num_phones $alidir/tree \
       "ark:gunzip -c $alidir/ali.*.gz | ali-to-phones $alidir/final.mdl ark:- ark:- |" \
-       $dir/0.ctc_trans_mdl || exit 1;
+       $dir/0.JOB.ctc_trans_mdl || exit 1;
+  for i in `seq 1 $num_cutoffs`;do
+    trans_model=$dir/0.${i}.ctc_trans_mdl
+    echo -n "$trans_model "
+    ctc-transition-model-info --print-args=false $dir/0.${i}.ctc_trans_mdl | grep num-history-states
+  done | python -c "
+import sys
```

Comment:
I find it slightly painful to see 10 lines of python when 1 line of awk would do.  How about testing something like this:
 awk -v t=$target_history_states '{if (NR==1 || ($2-t) < max_err) { best=NR; max_err=$2-2; }} END{print best;}'